### ISSUE TITLE
test: add 15 edge-case tests for agent tool and sampler

### ISSUE THREAD

**Issue by devswha** [devswha/zipcode#137]
## Summary

Autonomous QA audit identified undertested edge-case code paths in two critical modules. Added 15 edge-case unit tests covering boundary conditions, input validation, and numerical stability.

## Motivation

Analysis of test-to-code ratio across the workspace:
- `crates/tools/src/agent.rs`: 264 LOC / 7 tests → now 15 tests
- `crates/inference/src/sampler.rs`: 414 LOC / 14 tests → now 21 tests

These files handle critical runtime paths (agent delegation and token sampling) but had gaps in boundary condition testing.

### `crates/tools/src/agent.rs` (+8 tests)

| Test | What it verifies |
|------|-----------------|
| `test_depth_zero_succeeds` | Agent at depth 0 successfully delegates |
| `test_depth_one_succeeds` | Agent at depth 1 successfully delegates (boundary: MAX_AGENT_DEPTH=2) |
| `test_non_string_task_errors` | Integer `task` field returns error (not string) |
| `test_empty_task_accepted` | Empty string task is forwarded to callback |
| `test_allowlist_only_agent_stripped_to_none` | Allowlist `["agent"]` strips to empty vec |
| `test_valid_max_tokens_forwarded` | Integer `max_tokens` correctly forwarded to spawn callback |
| `test_no_max_tokens_passes_none` | Absent `max_tokens` passes `None` to callback |
| `test_schema_includes_optional_properties` | Schema contains `tool_allowlist`, `skill`, `max_tokens` properties |

### `crates/inference/src/sampler.rs` (+7 tests)

| Test | What it verifies |
|------|-----------------|
| `test_repeat_penalty_token_out_of_range_safe` | Token index > vocab size is silently skipped (no panic) |
| `test_temperature_exactly_one_no_scaling` | `temperature=1.0` is treated as no-op (no scaling applied) |
| `test_top_k_equals_vocab_size_no_filtering` | `top_k == vocab_size` passes all tokens through |
| `test_top_k_larger_than_vocab_disabled` | `top_k > vocab_size` disables filtering |
| `test_softmax_numerical_stability_large_logits` | Logits of 10000+ don't cause float overflow in softmax |
| `test_sequential_samples_with_different_history` | Multi-step sampling with penalty correctly tracks history |
| `test_all_negative_logits_with_temperature` | All-negative logits with temp > 0 samples correctly |

## Test Results

```
All 1,099 tests pass (0 failed, 2 pre-existing ignored)
- zipcode-tools: 186 passed (+8 new)
- zipcode-inference: 265 passed (+7 new)
- 0 clippy warnings
```

Commit: `685ca9f` on `main`

## Notes

- `omx exec --madmax` delegation was attempted but failed (401 Unauthorized — OpenAI auth issue)
- Work was completed directly using local tools per the fallback workflow
- One pre-existing intermittent failure observed: `server_options_env_overrides_flash_attention` — env var race condition in parallel tests (unrelated to this change)

**Comment by devswha** [comment]
Verified shipped on current main: `git log --oneline 685ca9f -1` → `test: add 15 edge-case tests for agent tool and sampler`. Title matches issue verbatim. Closing.
